File: booking_service/src/core/domain/dto/response/verify_promotion_result.go
package response

type VerifyPromotionResponse struct {
	IsValid    bool                     `json:"isValid"`
	Promotions []*VerifyPromotionResult `json:"promotions"`
}

type VerifyPromotionResult struct {
	PromotionID int64  `json:"promotionId"`
	IsValid     bool   `json:"isValid"`
	Message     string `json:"message"`
}

File: promotion_service/PromotionService.Core/Domain/Dto/Response/VerifyPromotionResult.cs
namespace PromotionService.Core.Domain.Dto.Response
{
    public class VerifyPromotionResult
    {
        public long PromotionId { get; set; }
        public bool IsValid { get; set; }
        public string Message { get; set; }
    }

    public class VerifyPromotionResponse
    {
        public List<VerifyPromotionResult> Promotions { get; set; }
        public bool IsValid { get; set; }
    }
}